homeassistant/components/binary_sensor/nest.py

```diff
@@ -60,11 +61,8 @@
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup Nest binary sensors."""
-    if discovery_info is None:
-        return
-
     nest = hass.data[DATA_NEST]
-    conf = config.get(CONF_MONITORED_CONDITIONS, _VALID_BINARY_SENSOR_TYPES)
```

Comment:
I'm pretty sure in the past this default would always have been used -- Using load_platform, config is *never* passed to the setup_platform method - see [here](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity_component.py#L83)

So the PR's statement 2 is because of the config Not being passed in...

The while idea if having validated schema under the platform AND using discovery only is conflicting in the current nest implementation

The intended design for load_platform was always to have the user config only on the main component, not the platforms! So I would suggest moving the config to the component level rather and using load_platform appropriately

The return if None statement should as a matter of fact fraise a warning stating to not do the config ont he platform level